throttling/src/test/java/com/iluwatar/tls/B2BServiceTest.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * The MIT License
+ * Copyright (c) 2014 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.iluwatar.tls;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.nio.file.AccessDeniedException;
+
+/**
+ * B2BServiceTest class to test the B2BService
+ */
+public class B2BServiceTest {
+
+  @Test
+  public void counterResetTest() throws AccessDeniedException {
+    Tenant tenant = new Tenant("testTenant", 100);
+    B2BService service = new B2BService(tenant);
+
+    for (int i = 0; i < 20; i++) {
+      service.dummyCustomerApi();
+      try {
+        Thread.sleep(100);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+    }
+    int counter = service.getCurrentCallsCount();
+    Assert.assertTrue("", counter < 11);
```

Comment:
Where this magic number 11 comes from?

Reply:
sleep for 100 ms means counter can go till 10. Hence the magic number :D 
I can do <= 10 if 11 is looking odd.